File: python/693. Binary Number with Alternating Bits.py
class Solution:
    def hasAlternatingBits(self, n: int) -> bool:
        prev=n&1
        n=n>>1
        while n>0:
            curr=n&1
            if prev==curr:
                return False
            else:
                prev=curr
                n=n>>1
        return True

File: java/693. Binary Number with Alternating Bits.java
class Solution {
    public boolean hasAlternatingBits(int n) {
        while(n!=0){
        int b=n&1;
        n >>>= 1;
        int c=n&1;
        if (b==c){
            return false;
        }

        }
        return true;


    }
}
